nengo/connection.py

```diff
@@ -319,6 +319,28 @@ def repeated_inds(x):
         return super().coerce(conn, transform)
 
 
+class ConnectionInitialValueParam(DistOrArrayParam):
+    """Connection-specific validation for initial values."""
+
+    def coerce(self, conn, value):
+        value = super().coerce(conn, value)
+
+        if isinstance(value, np.ndarray):
+            if isinstance(conn.post_obj, Ensemble) and conn.solver.weights:
+                n_synapses = conn.post_obj.n_neurons
+            else:
+                n_synapses = conn.size_out
+
+            if value.ndim > 0 and value.shape != (n_synapses,):
```

Comment:
I think `value.ndim` has to be `> 0`? Since there is `sample_shape=("*",)` the parent class seems to require that the array is at least 1D. I tried `initial_value=1` and got `nengo.exceptions.ValidationError: Connection.initial_value: ndarray must be 1D (got 0D)`. Unsure if this is intentional (would we want to support something like `initial_synapse=1` or would we require `initial_synapse=Choice([1])` or `initial_synapse=[1]`?

Reply:
So I flip-flopped on whether to allow a scalar to initialize all synapses to the same value. My final consensus was to not allow it, because you can accomplish the same thing with `Choice([0])` (it is a bit more verbose, but it fits more with how we typically use `DistOrArrayParam` and makes it a bit clearer what the user is asking for). That said, it's easy to allow it (all we need to do is remove the `sample_shape` when we instantiate this param, and I think it will already have the validation it needs due to this line here).

So thoughts on whether to allow it @arvoelke @tbekolay @drasmuss?

If we don't allow it, then we should remove the vestiges of when it was allowed that you've noted here, @arvoelke (namely this line, the docstring, and there's a bit of stuff in `builder/processes.py` I think).

Reply:
Based on a discussion we had on the Nengo dev team, I have now moved the `initial_value` parameter to `Synapse`, and called it `initial_output` instead. Scalars are now accepted (as is any array that is broadcastable to the required shape). This is in line with how things work for the new `NeuronType.initial_state`.